realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -91,7 +91,7 @@ public void onError(SyncSession session, ObjectServerError error) {
         }
     };
     // keeps track of SyncSession, using 'realm_path'. Java interface with the ObjectStore using the 'realm_path'
-    private static Map<String, SyncSession> sessions = new HashMap<String, SyncSession>();
+    private static Map<String, SyncSession> sessions = new ConcurrentHashMap<>();
```

Comment:
Even the newly added `notifyProgressListener` is synchronized on `SyncManager.class`, why do you need to use `ConcurrentHashMap`?